ring_buffer.go

```diff
@@ -12,28 +12,31 @@ var ErrIsEmpty = errors.New("ring buffer is empty")
 
 // RingBuffer 自动扩容循环缓冲区
 type RingBuffer struct {
-	buf     []byte
-	size    int
-	vr      int
-	r       int // next position to read
-	w       int // next position to write
-	isEmpty bool
+	buf      []byte
+	initSize int
+	size     int
+	vr       int
+	r        int // next position to read
+	w        int // next position to write
+	isEmpty  bool
 }
 
 // New 返回一个初始大小为 size 的 RingBuffer
 func New(size int) *RingBuffer {
 	return &RingBuffer{
-		buf:     make([]byte, size, size),
-		size:    size,
-		isEmpty: true,
+		buf:      make([]byte, size),
```

Comment:
We need to specify the `cap` or there will be a problem, we use `buf` as an array。